### PR TITLE
[HxGrid] A way to persist grid state (sorting, current page) ? #360

### PR DESCRIPTION
**Pull request by jirikanda**

### CODE REVIEW

**Comment by jirikanda** [comment]
Next step: Rename SortingItem<TItem> to GridSortingItem<TItem> while it contains grid specific property (SortDefaultOrder).

**Comment by jirikanda** [comment]
Work in progress...
TODO: Unit test, naming (and small refactorings).
(Renaming SortingItem to GridSortingItem not required anymore, SortDefaultOrder property removed.)

File: Havit.Blazor.Components.Web.Bootstrap/Grids/HxGridColumnBase.cs [hunk 1]
```diff
@@ -38,7 +41,16 @@ protected override void OnInitialized()
...
+	SortingItem<TItem>[] IHxGridColumn<TItem>.GetSorting() => sorting ??= this.GetSorting().ToArray();
+	private SortingItem<TItem>[] sorting;
+
+	/// <inheritdoc />
+	int? IHxGridColumn<TItem>.GetDefaultSortingOrder() => GetDefaultSortingOrder();
+
+	/// <summary>
+	/// Returns the unique column identifier.
+	/// </summary>
+	protected abstract string GetColumnIdentifier();
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Grids/HxGridColumnBase.cs:53]: @jirikanda `GetColumnId()` to align with the `Id` parameter?

File: Havit.Blazor.Components.Web.Bootstrap/Grids/IHxGridColumn.cs [hunk 2]
```diff
@@ -5,6 +5,11 @@
 /// </summary>
 public interface IHxGridColumn<TItem>
 {
+	/// <summary>
+	/// Returns the unique column identifier.
+	/// </summary>
+	string GetIdentifier();
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Grids/IHxGridColumn.cs:11]: `GetId()`?

**Review by hakenr — CHANGES_REQUESTED** [review]
lgtm, naming adjustments only

File: Havit.Blazor.Components.Web.Bootstrap/Grids/HxGrid.razor.cs [hunk 3]
```diff
@@ -250,14 +251,18 @@ public partial class HxGrid<TItem> : ComponentBase, IDisposable
 	[Inject] protected IStringLocalizer<HxGrid> HxGridLocalizer { get; set; }
 
 	private List<IHxGridColumn<TItem>> columnsList;
+	private HashSet<string> columnIds;
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Grids/HxGrid.razor.cs:254]: @jirikanda Not being used

**Review by hakenr — APPROVED** [review]
lgtm, merging

**Pull request merged**